pylibgen/pylibgen.py

```diff
@@ -143,8 +143,8 @@ def lookup(
             # TODO: add test case for this and see if i can .json() afterwards
             raise requests.HTTPError(400)
 
-        for book_data, _id in zip(resp, ids):
-            assert book_data["id"] == _id
+        for book_data in resp:
+            assert book_data["id"] in ids, "Invalid ids returned"
```

Comment:
What you've removed here is the precondition that libgen's api will return book data in order of the ids that were requested. I'm trying to assert for this, since libgen's api isn't stable.